src/Node.php

```diff
@@ -625,6 +628,7 @@ public function getNamespaceDefinition() {
             $namespaceDefinition = null;
         }
 
+        /** @var NamespaceDefinition|null $namespaceDefinition */
         return $namespaceDefinition;
```

Comment:
we could add a runtime check for stuff like this or refactor but it's probably a risky move considering the number of similar changes needed across the whole code base.